fuzzy/verifier.go

```diff
@@ -45,7 +45,13 @@ func (v *appendEntriesVerifier) PreRequestVote(src, target string, rv *raft.Requ
 
 func (v *appendEntriesVerifier) PreAppendEntries(src, target string, req *raft.AppendEntriesRequest) (*raft.AppendEntriesResponse, error) {
 	term := req.Term
-	ldr := string(req.Leader)
+	var ldr string
+	if req.ProtocolVersion > 3 {
+		ldr = string(req.Addr)
```

Comment:
Since this field is in an embedded structs we can access it this way, but generally I always try to be more explicit. I think including the embedded struct in the field access helps make it clear where the value is coming from:
```suggestion
		ldr = string(req.RPCHeader.Addr)
```

There are a few more places this is accessed, but I'll leave it to you if you think this is better or not in general.

Reply:
Yes good point, I always struggle to find all the references to an embedded field in my IDE because of that.